src/dataflow/src/render/sources.rs

```diff
@@ -435,11 +435,57 @@ where
                         // render envelopes
                         match &envelope {
                             SourceEnvelope::Debezium(dbz_envelope) => {
-                                let (stream, errors) = super::debezium::render(
-                                    dbz_envelope,
-                                    &results,
-                                    dataflow_debug_name.clone(),
-                                );
+                                // XXX(chae): use timestamps of END tx messages as timestamp for everything in the transaction
+                                // XXX(chae): use dbz deduplication full for tx source
+                                // XXX(chae): add depends_on for Sources in catalog
+                                let (stream, errors) = match dbz_envelope.tx_metadata {
+                                    Some(tx_id) => {
+                                        let tx_src_desc = storage_state
+                                            .source_descriptions
+                                            .get(&tx_id)
+                                            // N.B. tx_id is validated when constructing dbz_envelope
+                                            .expect("bad tx metadata spec")
+                                            .clone();
+                                        let ((tx_source_ok, tx_source_err), tx_token) =
+                                            import_source(
+                                                dataflow_debug_name,
+                                                dataflow_id,
+                                                as_of_frontier,
+                                                SourceInstanceDesc {
```

Comment:
is this the first example of making multiple source instances when we make a source instance?
Also, does using the same as_of_frontier make sense?

Reply:
Not sure if it's the first but that is what's happening here yeah.  Will def be a TODO to fix in future work!